include/awkward/array/NumpyArray.h

```diff
@@ -113,6 +113,38 @@ namespace awkward {
     const ssize_t itemsize_;
     const std::string format_;
   };
+
+  template <typename T>
+  const std::shared_ptr<Content> numpyarray_onedim(const IndexOf<T> count) {
+    std::vector<ssize_t> shape({ (ssize_t)count.length() });
+    std::vector<ssize_t> strides({ (ssize_t)sizeof(T) });
+    std::string format;
+  #ifdef _MSC_VER
+    if (std::is_same<T, int32_t>::value) {
+      format = "l";
+    }
+    else if (std::is_same<T, uint32_t>::value) {
+      format = "L";
+    }
+    else if (std::is_same<T, int64_t>::value) {
+      format = "q";
+    }
+  #else
+    if (std::is_same<T, int32_t>::value) {
+      format = "i";
+    }
+    else if (std::is_same<T, uint32_t>::value) {
+      format = "I";
+    }
+    else if (std::is_same<T, int64_t>::value) {
+      format = "l";
+    }
+  #endif
+    else {
+      throw std::runtime_error("unrecognized NumpyArray specialization");
+    }
+    return std::make_shared<NumpyArray>(Identities::none(), util::Parameters(), count.ptr(), shape, strides, 0, sizeof(T), format);
+  }
```

Comment:
At one point, I scanned through all the files and moved all implementations to the `.cpp` files, and in retrospect, I think that was a good thing. `numpyarray_onedim` doesn't need to be a free-standing function: how about if it's a constructor (or set of constructors, one for each `Index` type: `Index8`, `IndexU8`, `Index32`, `IndexU32`, `Index64`)?

If there's some technical problem with that, perhaps it could be a static function in `NumpyArray`. The only reason I make alternate constructors like this `staticmethods` or `classmethods` in Python is because Python doesn't have a dispatch mechanism for different argument types.

Reply:
@jpivarski - a constructor it is.